### ISSUE TITLE
test_jobcontroller_hworld.py regression test fails occassionally

### ISSUE THREAD

**Issue by jmlarson1** [Libensemble/libensemble#50]
I sometimes see the following when running

```
$ mpiexec -np 4 python3.5 test_jobcontroller_hworld.py
...
...
Expecting: [34 34 34 31 31 31 32 32 32 33 33 33  0  0  0]
Received:  [34 34 34 31 31 31 32 32 32 33 33  0 33  0  0]
```

It's intermittent, but occurs on my laptop and on Travis. Rerunning has always removed the error.

**Comment by jlnav** [comment]
This issue can be reproduced consistently on the Travis osx build jobs, although it is intermittent across combinations of number of processes and communication types.

**Comment by jlnav** [comment]
(Presumably) closed by #206
